File: go/gin/internal/modules/account/controller.go
package account

import (
	"net/http"
	"tcc/internal/common/errors"
	"tcc/internal/common/helper"
	"tcc/internal/common/messages"
	"tcc/internal/modules/account/dto"

	"github.com/gin-gonic/gin"
)

type AccountController struct{
	service AccountService
}

func (c *AccountController) FindAll(ctx *gin.Context){
	cpf := ctx.Query("cpf")

	dto := dto.FindAllQueryAccountDto{
		CPF: cpf,
	}

	// Transformações Necessárias
	dto.Transform()

	if err:= helper.Validate(dto); err != nil {
		badRequest := errors.BadRequestError(messages.ErrorMessages.BadRequest, err...)
		ctx.JSON(badRequest.StatusCode, badRequest)
		return 
	}


	client, err := c.service.FindAll(dto.CPF)
	if err != nil {
		ctx.JSON(err.StatusCode, err)
		return
	}

	ctx.JSON(http.StatusOK, client)
}

func (c *AccountController) Create(ctx *gin.Context){
	auth := helper.GetAuth(ctx)

	var dto dto.CreateAccountDto
	if err := ctx.ShouldBindJSON(&dto); err != nil {
		badRequest := errors.BadRequestError(messages.ErrorMessages.BadRequest, err.Error())
		ctx.JSON(badRequest.StatusCode, badRequest)
		return
	}
	
	if err:= helper.Validate(dto); err != nil {
		badRequest := errors.BadRequestError(messages.ErrorMessages.BadRequest, err...)
		ctx.JSON(badRequest.StatusCode, badRequest)
		return 
	}


	_, err := c.service.Create(dto, auth.ID, *auth)

	if err != nil {
		ctx.JSON(err.StatusCode, err)
		return
	}

	ctx.Status(http.StatusCreated)
}

func (c *AccountController) FindByCPF(ctx *gin.Context) {
	id := ctx.Param("id")

	auth := helper.GetAuth(ctx)

	dto := dto.FindOneAccountDto{
		ID: id,
	}

	if err:= helper.Validate(dto); err != nil {
		badRequest := errors.BadRequestError(messages.ErrorMessages.BadRequest, err...)
		ctx.JSON(badRequest.StatusCode, badRequest)
		return 
	}


	account, err := c.service.FindById(id, auth.ID, auth.Role)
	
	if err != nil {
		ctx.JSON(err.StatusCode, err)
		return
	}

	// Omitindo Campos não necessários
	account.IDAgencia = nil
	account.IDCliente = nil

	ctx.JSON(http.StatusOK, account)
}

File: rust/actix/src/modules/account/controller.rs
use actix_web::{web, HttpResponse, Responder};
use actix_web::web::Json;
use validator::Validate;
use crate::{common::errors::bad_request::bad_request_error, modules::auth::interfaces::JwtPayload};
use crate::common::messages::error::ErrorMessages;
use crate::common::errors::base::BaseError;
use super::service::AccountService;
use super::dto::{
    FindAllQueryAccountDto,
    CreateAccountDto,
    FindOneAccountDto,
};

pub struct AccountController {
}

impl AccountController {
    pub async fn find_all(
        mut dto: web::Query<FindAllQueryAccountDto>,
    ) -> Result<impl Responder, BaseError> {
        dto.transform();

        if let Err(err) = dto.validate() {
            return Err(bad_request_error(ErrorMessages::BAD_REQUEST, Some(err)));
        }

        match AccountService::find_all(&dto.cpf).await {
            Ok(accounts) => Ok(Json(accounts)),
            Err(err) => Err(err),
        }
    }

    pub async fn create(
        auth: web::ReqData<JwtPayload>,
        dto: web::Json<CreateAccountDto>,
    ) -> Result<impl Responder, BaseError> {
        if let Err(err) = dto.validate() {
            return Err(bad_request_error(ErrorMessages::BAD_REQUEST, Some(err)));
        }

        match AccountService::create(dto.into_inner(), &auth.id, auth.clone().into_inner()).await {
            Ok(_) => Ok(HttpResponse::Created().finish()),
            Err(err) => Err(err),
        }
    }

    pub async fn find_by_id(
        auth: web::ReqData<JwtPayload>,
        id: web::Path<String>,
    ) -> Result<impl Responder, BaseError> {
        let dto = FindOneAccountDto { id: id.into_inner() };

        if let Err(err) = dto.validate() {
            return Err(bad_request_error(ErrorMessages::BAD_REQUEST, Some(err)));
        }

        match AccountService::find_by_id(&dto.id, &auth.id, auth.role).await {
            Ok(mut account) => {
                // Omitindo campos não necessários
                account.id_agencia = None;
                account.id_cliente = None;
                Ok(Json(account))
            },
            Err(err) => Err(err),
        }
    }
}
